sys/include/seq.h

```diff
@@ -0,0 +1,493 @@
+
+/**
+ * @brief       A 8 bit sequence number
+ */
+typedef uint8_t seq8_t;
+
+/**
+ * @brief       A 16 bit sequence number
+ */
+typedef uint16_t seq16_t;
+
+/**
+ * @brief       A 32 bit sequence number
+ */
+typedef uint32_t seq32_t;
+
+/**
+ * @brief       A 64 bit sequence number
+ */
+typedef uint64_t seq64_t;
+
+/**
+ * @brief           Addition of a 8 bit sequence number @p s and a positive integer @p n
+ *                  in the serial number @p space
+ * @see             <a href="https://tools.ietf.org/html/rfc1982#section-3.1">
+ *                      3.1. Addition
+ *                  </a>
+ * @param[in]       s, sequence number
+ * @param[in]       n, positive integer in the range of [0 .. ((@p space / 2) - 1)]
+ * @param[in]       space, serial number space must be a power of 2 minus 1
+ * @return          s + n, if valid
+ * @return          s, if n is out of range
+ */
+static inline seq8_t seq8_adds(seq8_t s, uint8_t n, uint8_t space)
+{
+    if (n > SEQ_LIMIT(space)) {
+        return s;
+    }
+
+    return (space == UINT8_MAX) ? (s + n) : (s + n) % (space + 1);
+}
+
+/**
+ * @brief           Addition of a 8 bit sequence number @p s and a positive integer @p n
+ *                  in the serial number space UINT8_MAX
+ * @see             <a href="https://tools.ietf.org/html/rfc1982#section-3.1">
+ *                      3.1. Addition
+ *                  </a>
+ * @param[in]       s, sequence number
+ * @param[in]       n, positive integer in the range of [0 .. 127]
+ * @return          s + n, if valid
+ * @return          s, if n is out of range
+ */
+static inline seq8_t seq8_add(seq8_t s, uint8_t n)
+{
+    return seq8_adds(s, n, UINT8_MAX);
+}
+
+/**
+ * @brief           Increment a sequence number @p s by 1 in the serial number @p space
+ * @param[in]       s, sequence number
+ * @param[in]       space, serial number space must be a power of 2 minus 1
+ * @return          s + 1
+ */
+static inline seq8_t seq8_incs(seq8_t s, uint8_t space)
+{
+    return seq8_adds(s, 1, space);
+}
+
+/**
+ * @brief           Increment a sequence number @p s by 1 in the serial number space UINT8_MAX
+ * @param[in]       s, sequence number
+ * @return          s + 1
+ */
+static inline seq8_t seq8_inc(seq8_t s)
+{
+    return seq8_adds(s, 1, UINT8_MAX);
+}
+
+/**
+ * @brief           Compare sequence numbers @p s1, @p s2 in the serial number @p space
+ * @see             <a href="https://tools.ietf.org/html/rfc1982#section-3.2">
+ *                      3.2. Comparison
+ *                  </a>
+ * @param[in]       s1, first sequence number
+ * @param[in]       s2, second sequence number
+ * @param[in]       space, serial number space must be a power of 2 minus 1
+ * @return          -1, if s1 < s2
+ * @return          0, if s1 == s2
+ * @return          1, if s1 > s2
+ * @return          -EINVAL, if comparison of the pair (s1,s2) is undefined
+ */
+static inline int seq8_compares(seq8_t s1, seq8_t s2, uint8_t space)
```

Comment:
This and the other `_compares` functions are still pretty big for an inline function.